src/Serializer/DoctrineDenormalizer.php

```diff
@@ -63,7 +63,11 @@ public function denormalize($data, $class, $format = null, ?array $context = nul
 
             return $data;
         }
-
+        if (\is_string($data) === true) {
```

Comment:
What about integers or floats?

Reply:
I'll add integers - but I don't think we should be accepting floats as an index value

Reply:
Fixed.